server/test/src/main/java/org/eclipse/lsp/cobol/test/engine/AnnotatedDocumentCleaning.java

```diff
@@ -71,18 +70,37 @@ public PreprocessedDocument prepareDocument(
             explicitCopybooks.stream()
                 .findFirst()
                 .map(CobolText::getDialectType)
-                .orElse(null));
-
-    return new PreprocessedDocument(
-        testData.getText(),
-        processCopybooks(
-            collectCopybooks(explicitCopybooks, testData.getCopybookUsages(), sqlBackend, compilerOptions),
-            expectedDiagnostics,
-            testData),
-        testData);
+                .orElse(null), null);
+
+      List<CobolText> copybooks = collectCopybooks(explicitCopybooks, testData.getCopybookUsages(), sqlBackend, compilerOptions)
+              .map(c -> {
+                  TestData test = processDocument(
+                          c.getFullText(),
+                          c.getFileName(),
+                          toURI(c.getFileName(), c.getDialectType()),
+                          ImmutableList.of(),
+                          expectedDiagnostics,
+                          c.getDialectType(),
+                          testData.getCopybookEnterSectionNames().get(c.getFileName()));
```

Comment:
So what happens when I use the same copybook twice in different sections?

Reply:
It will blow up. Im not sure how intelligent and COBOL aware UseCase engine should be.